### PR TITLE
chore: update build.yml

### LINKED ISSUE #1894: Revisit GH Actions names
Some GH Action names are irrelevant (like `Unit Tests`, because it does execute integration tests as well).

### PR DESCRIPTION
**Pull request by thenswan**
**Description**:
Some GitHub Action names are irrelevant (like `Unit Tests`, because it does execute integration tests as well)-- Current PR addressed this issue.

Changes were inspired by [setup in Go SDK](https://github.com/hashgraph/hedera-sdk-go/blob/main/.github/workflows/build.yml).

**Related issue(s)**:
Closes #1894

**Notes for reviewer**:
`build.yml` files can be revisited in SDKs as a follow up.

### CODE REVIEW

**Review by isavov — APPROVED** [review]
lgtm

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -150,7 +150,7 @@ jobs:
       - name: Install Local Node
         run: npm install @hashgraph/hedera-local
 
-      - name: Compile Code
+      - name: Build SDK
         run: ./gradlew assemble --scan
 
       - name: Compile Code - Android Example
```
Comment (rbarkerSL) [inline .github/workflows/build.yml:156]: Should this `Compile Code - Android Example` be renamed to `Build Android Example` to be consistent with the rest of the name changes?

**Review by rbarkerSL — CHANGES_REQUESTED** [review]
One minor change requested for consistency.

Please create a github issue to update the repository settings for the required checks on PRs / pushes to main.

**Comment by rbarkerSL** [comment]
Please create a github issue to update the repository settings for the required checks on PRs / pushes to main.

After this PR merges the existing checks will not be valid (code compiles / example code tests / unit tests). We will need to add in checks the new named CI/CD items.

File: .github/workflows/build.yml (hunk 1 above)
Comment (thenswan) [reply .github/workflows/build.yml]: Agree, thanks @rbarkerSL. I updated it.

**Review by rbarkerSL — APPROVED** [review]

**Comment by thenswan** [comment]
@rbarkerSL thanks for review.
Created issue: https://github.com/hashgraph/hedera-sdk-java/issues/1903

**Pull request merged**